nixos/modules/services/cluster/kubernetes/dns.nix

```diff
@@ -205,24 +206,24 @@ in {
                 }
                 {
                   name = "sidecar";
-                  image = "gcr.io/google_containers/k8s-dns-sidecar-amd64:${version}";
-                  args = [
-                    "--v=2"
-                    "--logtostderr"
-                    "--probe=kubedns,127.0.0.1:10053,kubernetes.default.svc.${cfg.clusterDomain},5,A"
-                    "--probe=dnsmasq,127.0.0.1:53,kubernetes.default.svc.${cfg.clusterDomain},5,A"
-                  ];
+                  image = "k8s.gcr.io/k8s-dns-sidecar-amd64:${version}";
                   livenessProbe = {
-                    failureThreshold = 5;
                     httpGet = {
                       path = "/metrics";
                       port = 10054;
                       scheme = "HTTP";
                     };
                     initialDelaySeconds = 60;
-                    successThreshold = 1;
                     timeoutSeconds = 5;
+                    successThreshold = 1;
+                    failureThreshold = 5;
                   };
+                  args = [
+                    "--v=2"
+                    "--logtostderr"
```

Comment:
Motivation for logging to stderr is missing. For production services, I almost never want to do that.

Reply:
It's customnary in Docker containers: logs are then managed using logging drivers and aggregated using dedicated services. The alternative would be for the spawned process to write to files (as there's probably no logging daemon running in the container), but that approach brings up the issue of using a PersistentVolume just for the logs, which is far more invasive as a choice, in my opinion.

Reply:
@coretemp No personal preferences involved here. The yaml manifest is a copy of the upstream kube-dns manifest at: https://github.com/kubernetes/kubernetes/blob/release-1.10/cluster/addons/dns/kube-dns.yaml.base. A possible future improvement to the dns submodule could be to add more options for making these things configurable. For now though, I think it's perfectly fine to ship the upstream defaults.